plotnine/utils.py

```diff
@@ -1007,9 +1034,9 @@ def order_as_data_mapping(*args):
     elif n == 1:
         single_arg = ungroup(args[0])
         if isinstance(single_arg, pd.DataFrame):
-            return None, single_arg
-        elif isinstance(single_arg, aes):
             return single_arg, None
+        elif isinstance(single_arg, aes):
+            return None, single_arg
         else:
```

Comment:
The order of the return values here was flipped from the 2-arg-case. As far as I can see, everything but the 2 argument case is dead code, so that's why it never came up.

Reply:
The new version wouldn't pass the typechecker, because it couldn't prove that it would always return something. And in fact, if there were two mappings or two data arguments, then the code would just run through all the checks, never fail and return None. I've rewritten it so that can't happen.